### PR TITLE
Pull upstream changes

### PR DESCRIPTION
**Pull request by pellared**
## Why

Upstream sync 1fbb8c9..fad0ac5

## What

git log:

```
commit 1308dce5287eee07b3a82605e4803e4fbe35f439
Author: Lucas Pimentel-Ordyna <lucas.pimentel@datadoghq.com>
Date:   Thu Jan 6 13:03:56 2022 -0500

    more clean up in `SpanContextPropagator.Extract()` (#2261)
    
    * box values instead of allocating an array
    
    * remove unused `using` directives
    
    * refactor Extract(IReadOnlyDictionary<string, string>) as a call to Extract<T>(T, Func<...>)
    
    * replace yield return with new[]

commit 8ca3a378573fa53447900431860daa3798bb2470
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Thu Jan 6 08:28:18 2022 -0800

    [Test Package Versions Bump] (#2276)
    
    Co-authored-by: dependabot[bot] <dependabot[bot]@users.noreply.github.com>

commit 78ff87d7734b4b0ba8a3cd763427a919ab9eee7f
Author: Gregory LEOCADIE <gregory.leocadie@datadoghq.com>
Date:   Thu Jan 6 16:35:18 2022 +0100

    Fix InstallerVersion property. Set it to 2.0.1 (#2277)

commit 0dd81ae1eb8ff3c51a67d5814218c1ec00322b70
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Thu Jan 6 11:06:45 2022 +0000

    [Test Package Versions Bump] (#2246)
    
    Co-authored-by: dependabot[bot] <dependabot[bot]@users.noreply.github.com>

commit 6f2a01c6404a0c0b308b5a6693f8cff02f445224
Author: Andrew Lock <andrew.lock@datadoghq.com>
Date:   Wed Jan 5 14:31:17 2022 +0000

    Run GraphQL tests against latest package versions (#2264)
    
    * Run multi-version tests on GraphQL4
    
    * Update GraphQL4 tests to test against latest package versions
    
    Also ensure we test against .NET 6

commit c4f75fbc6c382b4c5ea2a43c29d1dd81e5a09fae
Author: Andrew Lock <andrew.lock@datadoghq.com>
Date:   Wed Jan 5 12:37:48 2022 +0000

    Add additional constructor to TracerSettings (#2266)
    
    * Add additional constructor to TracerSettings
    
    For discoverability purposes, since changes in version 2.0
    
    * Update public API
    
    * Update doc string

commit f1890de000e5c2c40aeaff2738aba0b6f23d12ce
Author: Alexander S <shurivich@gmail.com>
Date:   Wed Jan 5 14:13:14 2022 +0200

    Exploration Tests on Linux (#2193)
    
    Exploration test on Linux.
    Conditional trigger based on debugger/profiler changes for exploration tests.
    Variables generations on nuke.

commit b898802e77188115cd866afec5540f049e3c4066
Author: Lucas Pimentel-Ordyna <lucas.pimentel@datadoghq.com>
Date:   Tue Jan 4 15:38:55 2022 -0500

    clean up SourceGenerators and Analyzers projects (#2262)

commit 9f05fa028025fb434e00fb0ca2b5001dcbc163c7
Author: Kevin Gosse <krix33@gmail.com>
Date:   Tue Jan 4 14:37:48 2022 +0100

    Don't build CLI integration tests for net461 (#2263)
    
    * Don't build CLI integration tests for net461

commit a6aff8e343d0e6f18aa49f3ede09a68c910be918
Author: Kevin Gosse <krix33@gmail.com>
Date:   Mon Jan 3 23:33:16 2022 +0100

    Update autogenerated tags files (#2258)

commit 513a8535fbe657979f67eea9ff1cfc94aff0efbb
Author: Lucas Pimentel-Ordyna <lucas.pimentel@datadoghq.com>
Date:   Mon Jan 3 15:04:20 2022 -0500

    consolidate mock span implementations (#2119)
    
    * remove unused MsgPackHelpers and nuget packages
    
    use the vendor code instead
    
    * replace 3 implementation of mock spans with a new MockSpan
    
    * fix build errors
    
    * rename properties to match MessagePack schema
    
    and reduce PR diff
```

### CODE REVIEW

File: tracer/build/_build/Build.ExplorationTests.cs [hunk 1]
```diff
@@ -286,25 +329,25 @@ public static ExplorationTestDescription GetExplorationTestDescription(Explorati
...
-            ExplorationTestName.ilspy => new ExplorationTestDescription()
-            {
-                Name = "ILSpy",
-                GitRepositoryUrl = "https://github.com/icsharpcode/ILSpy.git",
-                GitRepositoryTag = "v7.1",
-                IsGitSubmodulesRequired = true,
-                PathToUnitTestProject = "ICSharpCode.Decompiler.Tests",
-                IsTestedByVSTest = true,
-                TestsToIgnore = new[] { "UseMc", "_net45", "ImplicitConversions", "ExplicitConversions", "ICSharpCode_Decompiler", "NewtonsoftJson_pcl_debug", "NRefactory_CSharp", "Random_TestCase_1", "AsyncForeach", "AsyncStreams", "AsyncUsing", "CS9_ExtensionGetEnumerator", "IndexRangeTest", "InterfaceTests", "UsingVariables" },
                 SupportedFrameworks = new[] { TargetFramework.NET461 },
```
Comment (Kielek) [inline tracer/build/_build/Build.ExplorationTests.cs:338]: here is something strange, you have commented out SupportedFrameworks = new[] { TargetFramework.NET461 }, exactly the same as uncommented.

Could you please double check?

File: tracer/test/test-applications/aspnet/Samples.AspNet.VersionConflict/Samples.AspNet.VersionConflict.csproj [hunk 2]
```diff
@@ -27,7 +27,7 @@
     <TargetFrameworkProfile />
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Datadog.Trace, Version=2.255.246.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb, processorArchitecture=MSIL">
+    <Reference Include="Datadog.Trace, Version=2.255.246.0, Culture=neutral, PublicKeyToken=e43a27c2023d388a, processorArchitecture=MSIL">
```
Comment (Kielek) [inline tracer/test/test-applications/aspnet/Samples.AspNet.VersionConflict/Samples.AspNet.VersionConflict.csproj:30]: are you sure about this change?

I know that there is no direct impact on us. Tests are disabled, but worth to check.

File: .azure-pipelines/ultimate-pipeline.yml [hunk 3]
```diff
@@ -104,6 +104,26 @@ resources:
 
 # Stages
 stages:
+- stage: generate_variables
```
Comment (Kielek) [inline .azure-pipelines/ultimate-pipeline.yml:107]: Do you have a plan to include this step also to our pipeline?

File: tracer/test/test-applications/aspnet/Samples.AspNet.VersionConflict/Samples.AspNet.VersionConflict.csproj (hunk 2 above)
Comment (pellared) [reply tracer/test/test-applications/aspnet/Samples.AspNet.VersionConflict/Samples.AspNet.VersionConflict.csproj]: You are correct. My bad. Reverted

File: tracer/build/_build/Build.ExplorationTests.cs (hunk 1 above)
Comment (pellared) [reply tracer/build/_build/Build.ExplorationTests.cs:338]: It simply looks as `ExplorationTestName.ilspy` was commented. It was done here: https://github.com/DataDog/dd-trace-dotnet/pull/2193

Comment (Kielek) [reply tracer/build/_build/Build.ExplorationTests.cs:338]: Sure, but paket previousle was targeted both to NET4.6.1 and NETCOREAPP#_1. Now only for NET4.6.1

Comment (pellared) [reply tracer/build/_build/Build.ExplorationTests.cs:338]: Yup. I propose to keep it as it is to be in sync with upstream.

File: .azure-pipelines/ultimate-pipeline.yml (hunk 3 above)
Comment (pjanotti) [reply .azure-pipelines/ultimate-pipeline.yml:107]: It seems worth but yeah, the related cs code on Nuke will require changes to work outside AZ pipelines.

**Review by pjanotti — APPROVED** [review]
LGTM

**Review by Kielek — APPROVED** [review]

**Pull request merged**